### PR TITLE
70 -> v1-release Card header padding

### PR DESCRIPTION
**Pull request by wmangimelli**
Screen Shot 2020-08-31 at 5 51 14 PM
Screen Shot 2020-08-31 at 5 52 29 PM
Screen Shot 2020-08-31 at 5 51 56 PM

### CODE REVIEW

File: packages/hs-react-ui/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -144,6 +144,10 @@ const Card = ({
   disableFeedback,
   feedbackType = FeedbackTypes.ripple,
 }: CardProps): JSX.Element | null => {
+  const hasContent = Boolean(children);
+  const hasFooter = Boolean(footer);
+  // eslint-disable-next-line no-param-reassign
+  headerProps = { ...headerProps, hasContent, hasFooter };
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Card/Card.tsx:150]: I'd just declare hasContent and hasFooter above the spread, and include them on the component declaration (Line 194) - and also pass both of those variables to the footer/body as well for people to utilize when writing their own styled components

File: packages/hs-react-ui/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -144,6 +144,10 @@ const Card = ({
   disableFeedback,
   feedbackType = FeedbackTypes.ripple,
 }: CardProps): JSX.Element | null => {
+  const hasContent = Boolean(children);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Card/Card.tsx:147]: would either call this `hasBody` or `hasChildren` to be specific to the nomenclature here 😄

Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.tsx:147]: Good call.

File: packages/hs-react-ui/src/components/Card/Card.tsx (hunk 1 above)
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.tsx:150]: On it!

Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.tsx:150]: @aVileBroker And should I go ahead and make a hasHeader to pass the Body and Footer?

File: packages/hs-react-ui/src/components/Card/Card.tsx (hunk 2 above)
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.tsx:147]: I will name it hasBody

File: packages/hs-react-ui/src/components/Card/Card.tsx (hunk 1 above)
Comment (aVileBroker) [reply packages/hs-react-ui/src/components/Card/Card.tsx]: Yeah I think that's a good idea

File: packages/hs-react-ui/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -51,11 +51,11 @@ export const CardContainer = styled(Div)`
 `;
 
 export const Header = styled(Div)`
-  ${() => {
+  ${({ hasBody, hasFooter }) => {
     const { colors } = useTheme();
 
     return `
-      padding: 1.5rem 1.5rem 0rem;
+      padding: 1.5rem 1.5rem ${!hasBody && !hasFooter ? '1.5rem' : '0rem'};
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Card/Card.tsx:58]: Could you add some snapshots to cover the cases where it has a body or a footer when there's a header? Maybe some of the other permutations

Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.tsx]: I add the some snapshot tests around this case. Also, I updated this statement to make it more clear.

**Review by aVileBroker — APPROVED** [review]

**Pull request merged**